src/pages/generate.tsx

```diff
@@ -188,28 +203,35 @@ export default function ClientIdentifierGenerator() {
     }
 
     // If the value has not initially been set, do not validate.
-    if (!form.getFieldMeta(fieldName).touched && !blurEvent.target.value) {
+    if (!formik.getFieldMeta(fieldName).touched && !blurEvent.target.value) {
       return;
     }
 
-    await validateFormField(fieldName, form.values);
+    await validateFormField(fieldName, formik.values);
 
     if (!formHasErrors()) {
       setHasFormError("");
     }
+
+    // Save state of the form.
+    sessionStorage.setItem("formValues", JSON.stringify(formik.values));
+    sessionStorage.setItem(
+      "formValidationState",
+      JSON.stringify(formik.states.all)
+    );
```

Comment:
These operations can throw, make sure they're handled safely.

Reply:
Addressed in https://github.com/inrupt/client-identifier-helper/pull/185/commits/ec0e24242114d647bbf77652364bc7891793ae7e